source/extensions/tracers/skywalking/tracer.cc

```diff
@@ -48,21 +48,24 @@ void Span::finishSpan() {
 
 void Span::injectContext(Tracing::TraceContext& trace_context,
                          const Upstream::HostDescriptionConstSharedPtr& upstream) {
-  const auto remote_address = upstream != nullptr ? upstream->address()->asString()
-                                                  : std::string(trace_context.authority());
-  // TODO(wbpcode): Due to https://github.com/SkyAPM/cpp2sky/issues/83 in cpp2sky, it is necessary
-  // to ensure that there is '\0' at the end of the string_view parameter to ensure that the
-  // corresponding trace header is generated correctly. For this reason, we cannot directly use host
-  // as argument. We need create a copy of std::string based on host and std::string will
-  // automatically add '\0' to the end of the string content.
-  auto sw8_header = tracing_context_->createSW8HeaderValue(remote_address);
+  absl::string_view remote_address =
+      upstream != nullptr ? upstream->address()->asStringView() : trace_context.authority();
+
+  auto sw8_header =
+      tracing_context_->createSW8HeaderValue({remote_address.data(), remote_address.size()});
```

Comment:
`createSW8HeaderValue` takes a `string_view`, so you can just pass `remote_address`?

Reply:
Yeah, I can. But my editor gives me an error `cannot convert absl::string_view to std::string`. 

May be my editor has some inappropriate local configurations. 🤔

Reply:
I get a compile error `candidate function not viable: no known conversion from 'absl::string_view' to 'std::string_view'`. But after C++17, these two types should be same type. 🤔

Reply:
@lizan Hah, I get it.

The `abseil.patch` set the `ABSL_OPTION_USE_STD_STRING_VIEW` to 0 make the `absl::string_view` an independent type rather than an alias of `std::string_view`.

It seems that we still need the code pattern in this PR to covert `absl::string_view` to `std::string_view`.